libzim/writer.py

```diff
@@ -183,8 +186,10 @@ def _get_counter_string(self):
         return ";".join(["%s=%s" % (k, v) for (k, v) in self._article_counter.items()])
 
     def close(self):
-        self.write_metadata()
-        self._creatorWrapper.finalize()
+        if hasattr(self, "_creatorWrapper"):
+            self.write_metadata()
+            self._creatorWrapper.finalize()
+            del self._creatorWrapper
```

Comment:
I'm not really confident with this `del`. It means that after a `close` the creator will be in an inconsistent state and a call to another method will lead to use a freed memory.

User should not call a method after a `close`, but we have no guaranty here.
If we delete the creator wrapper, we should at least test it before using it.

Or we can simply move the deletion in the `__del__` method.

Reply:
You're right. My idea was to prevent users from getting a `RuntimeError` when calling `.close()` twice because the user has no way to know whether the c_creator has been finalized or not which is somewhat problematic.

But this is a different topic so I'm removing that from here and will open a separate request to get that information.